### PR TITLE
fixed validate to async and use custamizeError function and editing s…

### PR DESCRIPTION
**Pull request by AmranElmasri**
fixed validate to async and use custamizeError function and editing somethins in frontend and everything is working properly

### CODE REVIEW

File: server/controllers/handleError.js [hunk 1]
```diff
@@ -2,13 +2,19 @@ const { join } = require('path');
 
 const clientError = (req, res)=>{
     res.status(404).sendFile(join(__dirname, '..', '..', 'public', 'html', '404.html'))
-}
+};
 
 
 const serverError = (err, req, res, next)=>{
-    console.log(err);
     res.status(500).sendFile(join(__dirname, '..', '..', 'public','html', '500.html'))
```
Comment (Raghad-Mezied) [inline server/controllers/handleError.js:9]: I think you missed this line, you are sending a response twice.

File: server/controllers/loginController.js [hunk 2]
```diff
@@ -1,47 +1,58 @@
...
+                id = data.rows[0].id
+                const { password: hashedPassword } = data.rows[0];
+                return compare(password, hashedPassword)
+            }
+        })
+        .then(isMatched => {
+            if (isMatched) {
+                sign({ id, email }, privateKey, (err, token) => {
+                    if (err) {
+                        res.status(500).json({ message: 'Internal server error' })
```
Comment (Raghad-Mezied) [inline server/controllers/loginController.js:35]: You can throw an error here like the others. The same for the other controllers handle their errors as you did here

**Review by Raghad-Mezied — COMMENTED** [review]
Great Amran 👏

File: server/controllers/handleError.js (hunk 1 above)
Comment (AmranElmasri) [reply server/controllers/handleError.js:9]: i will fix it

File: server/controllers/loginController.js (hunk 2 above)
Comment (AmranElmasri) [reply server/controllers/loginController.js:35]: yes sure i fixed it

**Pull request merged**